client/index.js

```diff
@@ -65,10 +66,27 @@ async function doRender ({ Component, props, err }) {
     props = await loadGetInitialProps(Component, { err, pathname, query })
   }
 
+  if (Component === ErrorComponent) {
+    lastScroll = {
+      x: window.pageXOffset,
+      y: window.pageYOffset
+    }
+  }
+
   Component = Component || lastAppProps.Component
   props = props || lastAppProps.props
 
   const appProps = { Component, props, err, router, headManager }
-  lastAppProps = appProps
   ReactDOM.render(createElement(App, appProps), container)
+
+  if (lastScroll &&
+    Component !== ErrorComponent &&
+    lastAppProps.Component === ErrorComponent) {
```

Comment:
Good catch. I removed it. Think the addition of `lastScroll` to the set check solved what I was trying to check before.